client/components/phone-input/index.jsx

```diff
@@ -48,19 +55,14 @@ const PhoneInput = React.createClass( {
 		};
 	},
 
-	componentWillReceiveProps( nextProps ) {
-		if ( this.props.value !== nextProps.value && nextProps.value ) {
-			const country = this.guessCountryFromValue( nextProps.value );
-			if ( country && country.isoCode !== this.getCountry().isoCode ) {
-				this.props.onChange( { countryCode: country.isoCode, value: nextProps.value } );
-			}
-		}
+	componentWillReceiveProps() {
+		// todo ensure the number is formatted if when loading from somewhere else
 	},
 
 	componentWillUpdate( nextProps ) {
-		const currentFormat = this.format( this.props ),
+		const currentFormat = this.props.value,
 			currentCursorPoint = this.numberInput.selectionStart,
-			nextFormat = this.format( nextProps );
+			nextFormat = nextProps.value;
 
 		let newCursorPoint = currentCursorPoint;
 		this.numberInput.value = nextFormat;
```

Comment:
Can't we just set `value={ this.props.value }` on the `input` (the `numberInput` one)? That should bind its value to `this.props.value`, without the need to update it manually here.